python/fapolicy_analyzer/tests/test_xdg_utils.py

```diff
@@ -0,0 +1,106 @@
+import os
+from util.xdg_utils import (
+    xdg_data_dir_prefix,
+    xdg_state_dir_prefix,
+    xdg_config_dir_prefix,
+)
+
+
+def test_xdg_state_dir_prefix_wo_env(monkeypatch):
+    if "XDG_STATE_HOME" in os.environ:
+        del os.environ["XDG_STATE_HOME"]
+
+    strBasename = "Arbitrary.txt"
+    strHomeDir = os.environ.get("HOME")
+    strStateDefaultDir = strHomeDir + "/.local/state/"
+    strExpected = strStateDefaultDir + "fapolicy-analyzer/" + strBasename
+
+    assert xdg_state_dir_prefix(strBasename) == strExpected
+
+
+def test_xdg_state_dir_prefix_w_env(monkeypatch):
+    strBasename = "Arbitrary.txt"
+    strEnvVar = "/tmp/"
+    strExpected = strEnvVar + "fapolicy-analyzer/" + strBasename
+
+    monkeypatch.setenv("XDG_STATE_HOME", strEnvVar)
+    strGenerated = xdg_state_dir_prefix(strBasename)
+    assert strGenerated == strExpected
+    assert os.path.exists(os.path.dirname(strGenerated))
+
+    # Clean-up
+    os.rmdir(os.path.dirname(strGenerated))
+    monkeypatch.delenv("XDG_STATE_HOME", raising=False)
+
+
+def test_xdg_state_dir_prefix_w_exception(monkeypatch):
+    strBasename = "Arbitrary.txt"
+    strEnvVar = "/usr/lib"  # Unwritable system dir
+    strExpected = "/tmp/" + strBasename
+
+    # Mock the environment to use a directory that can not be created
+    monkeypatch.setenv("XDG_STATE_HOME", strEnvVar)
+    assert xdg_state_dir_prefix(strBasename) == strExpected
+    monkeypatch.delenv("XDG_STATE_HOME", raising=False)
```

Comment:
Can you create a fixture called `env` to DRY some of this repeated code up?
```python
@pytest.fixture
def env(monkeypatch):
    monkeypatch.setenv("XDG_STATE_HOME", strEnvVar)
    yield
    monkeypatch.delenv("XDG_STATE_HOME", raising=False)
...
 def test_xdg_state_dir_prefix_w_exception(env):
    strBasename = "Arbitrary.txt"
    strEnvVar = "/usr/lib"  # Unwritable system dir
    strExpected = "/tmp/" + strBasename
    assert xdg_state_dir_prefix(strBasename) == strExpected
```

Reply:
I never did this before, but I think it would work.

Reply:
Other than the env var is set to two different directories to check a successful dir creation, and conversely to check dir creation failure to generate an exception. I don't know how to parameterize a fixture. I'll do some research...

Reply:
You can look at how I did the `confirm_dialog` fixture in `test_ancillary_trust_database_admin.py`.

Reply:
There doesn't appear to be a consensus or at least a simple one out in the community. Will continue to look because in the long run it will be useful for our overall unit=test suite.